geo/parse.go

```diff
@@ -383,23 +395,68 @@ func ExtractGeoShapeCoordinates(coordValue interface{},
 	typ string) ([][][][]float64, string, bool) {
 	var rv [][][][]float64
 	if typ == PointType {
-		rv = [][][][]float64{{{extractCoordinates(coordValue)}}}
+		point := extractCoordinates(coordValue)
+
+		// ignore the contents with invalid entry.
+		if len(point) < 2 {
+			return nil, typ, false
+		}
+
+		rv = [][][][]float64{{{point}}}
 		return rv, typ, true
 	}
 
 	if typ == MultiPointType || typ == LineStringType ||
 		typ == EnvelopeType {
-		rv = [][][][]float64{{extract2DCoordinates(coordValue)}}
+		coords := extract2DCoordinates(coordValue)
+
+		// ignore the contents with invalid entry.
+		if len(coords) == 0 {
+			return nil, typ, false
+		}
+
+		if len(coords) > 2 && typ == EnvelopeType {
+			return nil, typ, false
+		}
+
+		if len(coords) < 2 && typ != MultiPointType {
```

Comment:
`len(coords) < 2 && typ == MultiPointType` ?

Reply:
Ah in that case - I recommend updating the earlier check too :)

Reply:
This is to avoid 2 unnecessary checks for EnvelopeType when len(coords) < 2

Reply:
I realize the readability problem here. :)